### PR TITLE
Feat/finish tour form

### LINKED ISSUE #20: User Story:  Adding Tours
As a user, I should be able to add a tour, and view it in my dashboard
- [ ] Open app.
- [ ] Click +Tour
- [ ] Add a date and location, and submit the form 
- [ ] I should be returned to the home page, and see a new tour

### LINKED ISSUE #24: UI MVP: Add A Tour
I've opened this issue back up to track the next two pages of adding tour. Sorry for the confusion about what this issue included.

### PR DESCRIPTION
**Pull request by deadbelly**
### What Type of PR Is This?
- [x] New Feature 
- [ ] Bug Fix
- [ ] Testing
- [ ] Refactoring/Clean Up 

#### What does this branch do? 

- This branch adds in the second two pages to the trip form (RIDE, and DEBRIEF) and makes some general improvements to the form overall.
- Right now they still render vertically on the main page, I think it would be easiest to use routes to switch between them.
- Some of the fields are placeholders, for example the image asset on the RIDE page and the `group` text field. These give us something to work with while we wait to hook in actual back end data.
- Because of how I've programmed the dynamic rendering of text fields the primary controlled form state, `TourFormTextFields`, can only contain strings. Other data types (booleans, numbers, user objects) will need their own state variables.
- I think we may want to switch the input fields we use for notes to text boxes but I'm not positive. It should be easy to change all of them by editing `TextField.tsx` if we decide to make that switch. 
- I've also added a `util.tsx` file with a function that formats dates from date objects to `'YYYY-MM-DD'` strings. If we don't end up needing other helper files I can move this function somewhere else. 
- Right now there is no clear point at which the form should send a mutation request to the backend. In my opinion it would be easiest to send the data when the component dismounts either automatically, or by clicking a `SAVE` button. 
- The wire frames use static text fields with an `EDIT` option. I didn't implement this here, but again, if we want to do that we only need to edit `TextField.tsx`.
#### Have you tested your branch? How?
I've just tested it by rendering it in the DOM, manipulating inputs, and watching the state change. 
#### Do all existing unit and integration tests pass? 
N/A
#### Does this branch require special installation steps? 
Nope.
#### Related Issues:
closes #24  (for real this time)

### CODE REVIEW

File: src/components/Form/Plan.tsx [hunk 1]
```diff
@@ -1,33 +1,46 @@
 import React from 'react'
-import { TourFormState } from './TourForm'
 import { SectionTitle } from './SectionTitle'
-import { TextField } from './TextField'
-import './Form.css'
 
 interface PlanProps {
-  renderTextInputs: (fields: string[]) => JSX.Element[],
+  renderTextInputs: (fields: string[], prompts?: string[]) => JSX.Element[],
```
Comment (rachelbuchta) [inline src/components/Form/Plan.tsx:5]: what does prompts?: string[] mean here?

File: src/components/Form/TourForm.tsx [hunk 2]
```diff
@@ -37,39 +41,50 @@ export const TourForm: React.FC<TourFormProps> = ({ userId }) => {
     debriefPlan: ''
   })
 
-  const renderTextInputs = (fields: string[]): JSX.Element[] => {
-    return fields.map(field=> {
+  const [isDepartureChecked, setDepartureCheck] = useState<boolean>(false)
 
+  const renderTextInputs = (fields: string[], prompts?: string[]): JSX.Element[] => {
```
Comment (rachelbuchta) [inline src/components/Form/TourForm.tsx:46]: Okay, I think i'm getting it now. prompts is an optional parameter here.

File: src/components/Form/TourForm.tsx [hunk 3]
```diff
@@ -37,39 +41,50 @@ export const TourForm: React.FC<TourFormProps> = ({ userId }) => {
...
+          value={textFields[field as keyof TourFormTextFields]}
+          updateForm={e => setTextFields({ ...textFields, [field]: e.target.value})}
         />
       )
     })
   }
 
   const isChecked = (fields: string[]) => {
-    return !fields.find(field => state[field as keyof TourFormState] === '')
+    return !fields.find(field => textFields[field as keyof TourFormTextFields] === '')
```
Comment (rachelbuchta) [inline src/components/Form/TourForm.tsx:60]: I think you're going to have to talk me through the logic of this one here. I know what it's responsible for and what it's doing but i'm not entirely understanding how this logic works to make the check box toggle back and forth

File: src/components/Form/Ride.tsx [hunk 4]
```diff
@@ -1,10 +1,30 @@
 import React from 'react'
+import placeholder from '../../assets/placeholderChartAsset.png'
```
Comment (tashiad) [inline src/components/Form/Ride.tsx:2]: I'll try to find an actual asset for this that we can use

File: src/util.tsx [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const getDateString = (date: Date): string => {
+  return date.toISOString().substring(0, 10)
```
Comment (rachelbuchta) [inline src/util.tsx:2]: Are these built in javascript methods?! Tell me more!!

**Review by rachelbuchta — APPROVED** [review]

File: src/util.tsx (hunk 5 above)
Comment (tashiad) [inline src/util.tsx:2]: You figured it out! How does this work?

**Comment by rachelbuchta** [comment]
Booone!! You crushed this. Thanks for being so efficient in knocking this part out. This is pretty much the meat of this whole app so the fact that this is up and working is huge!

**Review by tashiad — APPROVED** [review]
Looks great! Thanks for taking this one on. One comment (doesn't have to happen now) but we'll need to add the save buttons to the end of each form. Or maybe we'll want each individual input to have an edit/save button? Rachel and I already have some similar logic written from Crate [here](https://github.com/isabellevillasenor/crate/blob/6961445dec613ba729482aadd6cec099c945f47b/code/web/src/modules/user/MyAccount.js)

**Pull request merged**